tpm2/structures.go

```diff
@@ -371,3 +371,158 @@ type tpmtSigScheme struct {
 	Scheme Algorithm
 	Hash   Algorithm
 }
+
+// AttestationData contains data attested by TPM commands (like Certify).
+type AttestationData struct {
+	Magic               uint32
+	Type                tpmutil.Tag
+	QualifiedSigner     Name
+	ExtraData           []byte
+	ClockInfo           ClockInfo
+	FirmwareVersion     uint64
+	AttestedCertifyInfo *CertifyInfo
```

Comment:
Why is QualifiedSigner (a Name struct) inline while AttestedCertifyInfo (a CertifyInfo) is a pointer?

Reply:
QualifiedSigner is always present in TPM response.
AttestedCertifyInfo is only populated if Type == tagAttestCertify. If we added support for other Type values, we'd add corresponding pointer Attested* fields. Only one of those fields would be set.